lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -597,22 +598,29 @@ spec = do
                     ]
 
         it "pools have the correct retirement information" $ \ctx -> do
-            eventually "pools have the correct retirement information" $ do
-                r <- listPools ctx arbitraryStake
-                expectResponseCode HTTP.status200 r
-                let retirementEpochIs n = \case
-                        Nothing -> False
-                        Just ei -> epochNumber ei == ApiT (EpochNo n)
-                verify r
-                    [ expectListSize 3
-                    , expectListField 0 #retirement
-                        (`shouldBe` Nothing)
-                    , expectListField 1 #retirement
-                        (`shouldSatisfy` retirementEpochIs 1_000)
-                    , expectListField 2 #retirement
-                        (`shouldSatisfy` retirementEpochIs 1_000_000)
+
+            let expectedRetirementEpochs = Set.fromList
+                    [ Nothing
+                    , Just 1_000
+                    , Just 1_000_000
                     ]
 
+            eventually "pools have the correct retirement information" $ do
+                response <- listPools ctx arbitraryStake
+                expectResponseCode HTTP.status200 response
+
+                let getRetirementEpoch :: ApiStakePool -> Maybe EpochNo
+                    getRetirementEpoch =
+                        fmap (view (#epochNumber . #getApiT))
+                        .
+                        view #retirement
+
+                let actualRetirementEpochs =
+                        getFromResponse Prelude.id response
```

Comment:
by the way: getFromResponse expects a lens as a first argument, so `getFromResponse #retirement` works just fine.

Reply:
> by the way: getFromResponse expects a lens as a first argument, so `getFromResponse #retirement` works just fine.

We're dealing with a list here though, so the expression `getFromResponse #retirement response` produces the error:

```
The type [ApiStakePool] does not contain a field named 'retirement'
```